CONTRIBUTING.md

```diff
@@ -1,65 +1,121 @@
-# How to contribute #
+Contributing to the Exiv2 Project
+======================
 
-## General workflow for contributing ##
+# Contents #
 
-1. Fork, then clone the repo (more details [here](https://guides.github.com/activities/forking/)):
-```
-git clone git@github.com:your-username/exiv2.git
-```
+* [1. Introduction](#1-introduction)
+* [2. General Workflow for Contributing Code](#2-general-workflow-for-contributing-code)
+* [3. Contributing Lens Data](#3-contributing-lens-data)
+* [4. Pull Request Guidelines](#4-pull-request-guidelines)
+* [5. Coding Guidelines](#5-coding-guidelines)
+  * [5.1 General](#51-general)
+  * [5.2 Integer and Array Overflows](#52-integer-and-array-overflows)
+  * [5.3 Code Formatting](#53-code-formatting)
+* [6. Reporting Bugs](#6-reporting-bugs)
 
-2. Configure the project and build (CMake is our the preferred configuration system, but not the only one. More details in the [README.md](README.md)) :
-```
-mkdir build && cd build
-cmake .. && cmake --build . --config Release
-```
+# 1. Introduction #
+
+This file describes how users and developers can contribute to the Exiv2 project. We welcome any help, like for example, contributing lens data (images), code contributions, and bug reports.
+In the past this was organized using Redmine at: [dev.exiv2.org](http://dev.exiv2.org). As of April 2017 the Exiv2 project has moved from svn and Redmine to GitHub and information, such
+as, Wikis etc., will gradually be moved from Redmine to GitHub. User requests on the Redmine forums and the Redmine issue tracking system will be serviced until Exiv2 0.27 is released. 
+After 0.27 is released the Redmine site will be read-only and all feature requests, bug reports etc., must be performed on GitHub. It is, however, preferred to use GitHub for all new 
+feature requests and bug reports to Exiv2.
+
+# 2. General Workflow for Contributing Code #
+
+Code contributions is performed by using *pull requests* (PR) on GitHub. For this to work one first needs to 
+[create a user account on GitHub](https://help.github.com/articles/signing-up-for-a-new-github-account/) if one do not already have one. Once you have a GitHub login:
+
+1. Fork the Exiv2 git repository to your GitHub account by pressing the _Fork_ button at: [https://github.com/Exiv2/exiv2](https://github.com/Exiv2/exiv2) (more details [here](https://guides.github.com/activities/forking/)).
+
+1. Then start a terminal (or use your favorite git GUI app) and clone your copy of the Exiv2 repo:
+
+        $  git clone https://github.com:your-username/exiv2.git
+
+2. Configure the project and build it (CMake is our the preferred configuration system, but not the only one [see [README.md](README.md)]):
+
+        $ cd exiv2
```

Comment:
Could you fix the indentation here, please? And it would be also good to use a block of code (delimited with three back-ticks)